ci/do_circle_ci.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -e
+
+# bazel uses jgit internally and the default circle-ci .gitconfig says to
+# convert https://github.com to ssh://git@github.com, which jgit does not support.
+mv ~/.gitconfig ~/.gitconfig_save
+
+export ENVOY_SRCDIR="$(pwd)"
+export NUM_CPUS=8 # xlarge resource_class
```

Comment:
Nit: why not let this be dynamically discovered to make it more robust?

Reply:
See the note here: https://circleci.com/docs/2.0/configuration-reference/#resource_class (It relates to how docker works unfortunately). I will add a comment here and generally clean this review up if we get IPv6 support.

Reply:
As @alyssawilk suggests on other threads, if we want to keep Travis around for just some IPv6 stuff, then we don't need to block the Circle CI move.